packages/x-components/src/x-modules/queries-preview/components/__tests__/query-preview.spec.ts

```diff
@@ -119,9 +120,24 @@ describe('query preview', () => {
     expect(wrapper.emitted('load')?.[0]).toEqual(['shoes']);
   });
 
+  it('does not remove the query before destroy the component', () => {
+    const { wrapper } = renderQueryPreview({
+      persistInCache: true,
+      queryPreviewInfo: {
+        query: 'shoes',
+        extraParams: { directory: 'Magrathea' },
+        filters: ['fit:regular']
+      }
+    });
+
+    wrapper.destroy();
+    expect(wrapper.emitted('load')?.length).toBe(1);
+    expect(wrapper.emitted('load')?.[0]).toEqual(['shoes']);
```

Comment:
here I think what should done is to check the state & see that there are still query preview there